### PR TITLE
[Easy] [Labels] Implement trader_dex_diversity label

### PR DESCRIPTION
**Pull request by gentrexha**
Brief comments on the purpose of your changes:

Adding a trader dex diversity label for addresses that have completed at least one trade on a dex or dex aggregator and counting the number of different projects they've used in the past 30 days.

cc @bh2smith for internal review

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
~~* [ ] if adding a new model, I added a test~~
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
~~* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)~~
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Cool label idea. So is the label an integer? under the category "dex diversity"?

**Comment by gentrexha** [comment]
> Cool label idea. So is the label an integer? Under the category "dex diversity"?

Thanks! The label is an integer from 1 to N, where N is the total number of projects in the dex and dex aggregator tables.

File: models/labels/trader_dex_diversity/ethereum/trader_dex_diversity_ethereum.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        from {{ ref('dex_trades') }}
+        where blockchain = 'ethereum' and block_time > (current_date - interval '30 days')
+    )
+    group by taker
+ )
+
+select
+  array("ethereum") as blockchain,
+  address,
+  dex_diversity as name,
```
Comment (soispoke) [inline models/labels/trader_dex_diversity/ethereum/trader_dex_diversity_ethereum.sql:27]: Maybe add: Number of DEXs traded on: `dex_diversity` ?

Comment (gentrexha) [reply models/labels/trader_dex_diversity/ethereum/trader_dex_diversity_ethereum.sql]: Let me know if the latest update to the name is what you had in mind.

Comment (soispoke) [reply models/labels/trader_dex_diversity/ethereum/trader_dex_diversity_ethereum.sql]: Yes looks good! Now I'm just thinking: do you think this info will be a bit redundant with the `dex_user` model in which we print the actual names of DEXs the address has traded on? It's not a big deal we could have both, just wanted to have your opinion on this

Comment (gentrexha) [reply models/labels/trader_dex_diversity/ethereum/trader_dex_diversity_ethereum.sql]: IMHO: Having just the number is better for visualizations as it indicates that a user uses multiple DEXs without mentioning which. Whereas, for having just the names, there are various combinations for users who use more than two DEXs and, therefore, hard to group by.

**Comment by gentrexha** [comment]
Thanks @soispoke !

**Pull request merged**